pkg/devfile/generator/utils.go

```diff
@@ -153,11 +153,20 @@ func addSyncRootFolder(container *corev1.Container, sourceMapping string) string
 
 	// Note: PROJECTS_ROOT & PROJECT_SOURCE are validated at the devfile parser level
 	// Add PROJECTS_ROOT to the container
-	container.Env = append(container.Env,
-		corev1.EnvVar{
+	container.Env = append([]corev1.EnvVar{
+		{
 			Name:  EnvProjectsRoot,
 			Value: syncRootFolder,
-		})
+		},
+	}, container.Env...)
```

Comment:
I was looking into the issue. The goal of the issue associated with this PR is to ensure that `PROJECT_ROOT` and `PROJECT_SOURCE` are the first 2 environment variables  in the env var list. The way this append is structured with the spread of `container.Env...` after should essentially be prepending the value to the front of the slice. I believe if we do that for both `PROJECT_ROOT` and `PROJECT_SOURCE` then we are good @thepetk

Reply:
Oh ok that makes sense. So my comment based on this, it might be better to check that the `PROJECT_ROOT` is on the top of the env vars list in [addSyncRootFolder](https://github.com/devfile/library/blob/64f49c75c68f6daf5e1b1cb26e2acd40eb57f8db/pkg/devfile/generator/utils.go#L146) (we're already doing that) and the `PROJECT_SOURCE` in the [addSyncFolder](https://github.com/devfile/library/blob/main/pkg/devfile/generator/utils.go#L168-L199).

That said I think this [block](https://github.com/devfile/library/blob/64f49c75c68f6daf5e1b1cb26e2acd40eb57f8db/pkg/devfile/generator/utils.go#L163-L169) should be removed, while this [block](https://github.com/devfile/library/blob/64f49c75c68f6daf5e1b1cb26e2acd40eb57f8db/pkg/devfile/generator/utils.go#L201-L205) should make sure that the env var is on the top of the env vars list, just like we're doing here.

Reply:
> I was looking into the issue. The goal of the issue associated with this PR is to ensure that `PROJECT_ROOT` and `PROJECT_SOURCE` are the first 2 environment variables in the env var list. The way this append is structured with the spread of `container.Env...` after should essentially be prepending the value to the front of the slice. I believe if we do that for both `PROJECT_ROOT` and `PROJECT_SOURCE` then we are good @thepetk

but currently `PROJECT_SOURCE` taking same value as `PROJECT_ROOT` . i was thinking as you said append the two first but we shouldnt appedn blindly , we can just check if the both have same values then it should be null .

Reply:
@Horiodino based on the comment I made above and @Jdubrick requested changes, I think our suggested workaround is not changing the way we append the env vars, it only makes sure that each time we append `PROJECT_SOURCE` and/or `PROJECT_ROOT` we make sure they are on the top.